src/ProjectManagement/Strings.resx

```diff
@@ -234,6 +234,9 @@
   <data name="RemovingPackageFromMSBuildProject" xml:space="preserve">
     <value>Removing package '{0}' from project '{1}'</value>
   </data>
+  <data name="SolutionDirectoryShouldBeANonNullFullPath" xml:space="preserve">
+    <value>Solution directory must be a non-null full path</value>
```

Comment:
This is never shown to the user in visual studio, this is an error of someone using the API. 

Check for null or empty, and provide an error saying it cannot be null

Then check for full path and provide an error saying the path must be rooted, and provide the path that was passed in.